### ISSUE TITLE
Advanced module settings

### ISSUE THREAD

**Issue by buddh4** [humhub/humhub#4198]
Currently module class level configurations can only be changed by configuration file. Some of those settings may be candidates for beeing available in the administration backend in a future release as well. It would be great to add an addional default check to `Yii::$app->getModule('xy')->settings->get()` as follows:

1. Search in db for given module setting
2. Return default value given by argument 
3. If no default argument is given, check for existing property in module class

This way we'd only use `$module->settings->get()` instead of directly checking the module class property which would also fix https://github.com/humhub/humhub/issues/3872

**Comment by buddh4** [comment]
Not sure about the order of step 2. and 3. Either the argument default or the module default should have priority. In most cases you won't use an argument default if you know there is a module class level default.

**Comment by luke-** [comment]
Nice idea! I think we should always prefer the default value given as argument. 

But we should describe this behaviour as clearly as possible. It is probably not so easy to understand where which values come from.

`Database (SettingsManager) > Argument Default > Module Class Property > Configuration File(s) Overwrite`

**Comment by buddh4** [comment]
You are right, as I said you probably won't use a default argument in most cases when you have a class level default.
